embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/internal/spans/EmbraceSpanFactoryImplTest.kt

```diff
@@ -18,13 +18,13 @@ internal class EmbraceSpanFactoryImplTest {
     private val clock = FakeClock()
     private lateinit var embraceSpanFactory: EmbraceSpanFactoryImpl
     private lateinit var spanRepository: SpanRepository
-    private lateinit var tracer: Tracer
+    private lateinit var tracer: FakeTracer
 
     @Before
     fun setup() {
         val initModule = FakeInitModule(clock)
         spanRepository = initModule.openTelemetryModule.spanRepository
-        tracer = initModule.openTelemetryModule.tracer
+        tracer = FakeTracer()
```

Comment:
Changing the code to check whether the associated span is recording before returning that a `start` call has succeeded means that if I used a real Tracer, it would find that the parent I had set is fake (and therefore the `Context` object isn't populated), so it'll fail.

Since the goal for this is testing the unit, I'm going to use a fake in here instead so the implementation detail of another object doesn't affect the results of this test.